### PR TITLE
[SchemaRegistry] overload get schema to take version

### PR DESCRIPTION
**Pull request by swathipil**
OOB discussion with Johan:
* Remove `get_schema_by_version` method from sync and async SchemaRegistryClient
* Add `get_schema` overload, which takes `group_name`, `name`, and `version` as keyword args.

Addressing: #25072
Follow up from: #26055

### CODE REVIEW

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py [hunk 1]
```diff
@@ -136,19 +138,36 @@ async def register_schema(
...
+    @overload
+    async def get_schema(
+        self, *, group_name: str, name: str, version: int, **kwargs
+    ) -> Schema:
+        ...
+
+    @distributed_trace_async
+    async def get_schema(self, *args: Union[str, int], **kwargs: Any) -> Schema:
+        """Gets a registered schema. There are two ways to call this method:
+        1) To get a registered schema by its unique ID, pass the `schema_id` parameter and any optional
```
Comment (JoshLove-msft) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py:161]: What optional keyword args are there when passing in schema ID?

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py [hunk 2]
```diff
@@ -136,19 +138,36 @@ async def register_schema(
...
+        ...
+
+    @overload
+    async def get_schema(
+        self, *, group_name: str, name: str, version: int, **kwargs
+    ) -> Schema:
+        ...
+
+    @distributed_trace_async
+    async def get_schema(self, *args: Union[str, int], **kwargs: Any) -> Schema:
```
Comment (JoshLove-msft) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py:159]: why are the args a Union of str and int? Can you pass schema Id or version as args?

**Review by JoshLove-msft — APPROVED** [review]
Just some questions that are probably a result of me not understanding Python 😆 
But based on the samples, it all looks good.

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py (hunk 1 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py:161]: so optional keywords are ones that are reserved by azure-core that users can [pass in for the request itself](https://github.com/Azure/azure-sdk-for-python/blob/b37efe5deb6aaa096f039a565e7222201257f0f8/sdk/core/azure-core/azure/core/rest/_rest_py3.py#L68). for ex, params/json/content/data.

Yep, both overloads would appear in the docs!

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py (hunk 2 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py:159]: ah good catch! I think args should just be str - will fix this. I added int here b/c version is an int. However, version will be passed in as a kwarg(`version=`) and not a positional argument, so we don't need the type of version to be included in args.

So schema_id can be passed in as a positional argument, (i.e. we don't require the `schema_id=` to be passed in) or as a keyword arg (i.e. schema_id=schema_id). If it's passed in as a positional arg, it'll be in `args`. If it's passed in as a kwarg, then the type will be included in `Any`.

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py [hunk 3]
```diff
@@ -162,50 +181,51 @@ def get_schema(self, schema_id, **kwargs):
...
         """
         http_request_kwargs = get_http_request_kwargs(kwargs)
-        request = schema_rest.build_get_schema_version_request(
-            group_name=group_name, schema_name=name, schema_version=version, **http_request_kwargs
-        )
+        try:
+            try:
+                schema_id = args[0]
+            except IndexError:
+                schema_id = kwargs.pop("schema_id")
```
Comment (deyaaeldeen) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py:196]: nitpick, is there a nicer way to do this access? like checking args' length first?

Comment (deyaaeldeen) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py:196]: also could readability be improved by factoring the schema_id logic into a helper function?

**Review by deyaaeldeen — APPROVED** [review]
🚀 happy to see all languages aligned

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py (hunk 3 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py:196]: In Python, we usually use the try/except pattern over the if/else. I can add comments for each block for more clarity?

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py [hunk 4]
```diff
@@ -162,50 +181,54 @@ def get_schema(self, schema_id, **kwargs):
...
+            )
+        except KeyError:
+            # If group_name, name, and version aren't passed in as kwargs, raise error.
+            try:
+                group_name = kwargs.pop("group_name")
+                name = kwargs.pop("name")
+                version = kwargs.pop("version")
+            except KeyError as exc:
+                raise TypeError(
+                    f"""If getting schema by version, '{exc.args[0]}' is a required keyword."""
```
Comment (johanste) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py:211]: We should make the error message look more like a "vanilla" TypeError you get if calling with mismatched parameters. And we should assume that it was the original method (by id) that was intended unless we believe that the "by version" method is more commonly used.

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_generated/aio/operations/_operations.py [hunk 5]
```diff
@@ -415,6 +431,7 @@ async def register(  # pylint: disable=inconsistent-return-statements
             401: ClientAuthenticationError,
             404: ResourceNotFoundError,
             409: ResourceExistsError,
+            304: ResourceNotModifiedError,
```
Comment (johanste) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_generated/aio/operations/_operations.py:434]: Is this a GET call? If not, why are we expecting a 304 answer? Is this just "let's apply the same map everywhere" thing?

Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_generated/aio/operations/_operations.py:434]: No, this is a [PUT call](https://github.com/Azure/azure-rest-api-specs/blob/4b2f5a45a34e351e38e8b075cf1da5a775fd414a/specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2021-10/schemaregistry.json#L385). According to Izzy and [[this issue](https://github.com/Azure/autorest.python/issues/1415)], this is added by default everywhere and doesn't take input from the swagger.

**Pull request merged**